### PR TITLE
Add journal storage loader

### PR DESCRIPTION
**Pull request by gen740**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

None

## What does this implement/fix? Explain your changes.

- Add `journalStorage.ts` into standalone app, enabling `journal storage` support.

### CODE REVIEW

**Comment by nabenabe0928** [comment]
@c-bata Could you review this PR?

**Comment by nabenabe0928** [comment]
@c-bata I removed you from the assignee of this PR.
@keisuke-umezawa Could you review this PR?

File: standalone_app/src/components/StorageLoader.tsx [hunk 1]
```diff
@@ -30,7 +31,13 @@ export const StorageLoader: FC = () => {
     r.addEventListener("load", () => {
       const arrayBuffer = r.result as ArrayBuffer | null
       if (arrayBuffer !== null) {
-        loadStorage(arrayBuffer, setStudies)
+        const header = new Uint8Array(arrayBuffer, 0, 16)
+        const headerString = new TextDecoder().decode(header)
+        if (headerString === "SQLite format 3\u0000") {
+          loadSQLite3Storage(arrayBuffer, setStudies)
+        } else {
+          loadJournalStorage(arrayBuffer, setStudies)
```
Comment (keisuke-umezawa) [inline standalone_app/src/components/StorageLoader.tsx:39]: [question] Do you have any way to check that the file is compatible with JournalStorage? If yes, we can add "else if" to check it.

File: standalone_app/src/vscode_entry.tsx [hunk 2]
```diff
@@ -35,7 +38,13 @@ export const AppWrapper: FC = () => {
             bytes[i] = binaryString.charCodeAt(i)
           }
           arrayBuffer = bytes.buffer
-          loadStorage(arrayBuffer, onceSetStudies)
+          header = new Uint8Array(arrayBuffer, 0, 16)
+          headerString = new TextDecoder().decode(header)
+          if (headerString === "SQLite format 3\u0000") {
+            loadSQLite3Storage(arrayBuffer, onceSetStudies)
+          } else {
+            loadJournalStorage(arrayBuffer, onceSetStudies)
```
Comment (keisuke-umezawa) [inline standalone_app/src/vscode_entry.tsx:46]: ditto.

**Review by keisuke-umezawa — COMMENTED** [review]
This is early review comment. I checked PR other than JournalStorage implementation. I will check it later 🙏 
Btw, I hope we can add some tests for it. At least, can you add e2e test for this one similar to this test?

https://github.com/optuna/optuna-dashboard/blob/38cc928df3423c08eb5dea60bd67d097d0e833b0/e2e_tests/test_standalone/test_study_list.py#L28-L74

File: standalone_app/src/components/StorageLoader.tsx (hunk 1 above)
Comment (gen740) [reply standalone_app/src/components/StorageLoader.tsx:39]: Unfortunately, there is no "header" or special "flag" in the JournalStorage log. The only requirement for JournalStorage is a JSON line file. We can detect the format by parsing the first line of the log, but I believe this should be handled in the `JournalStorage` class.

**Comment by gen740** [comment]
Thank you for your comments!
Currently, Optuna-dashboard's standalone app lacks a test system. Consequently, there is no way to implement tests at this time. However, I plan to develop and add a testing system for the standalone app as a follow-up.

**Comment by keisuke-umezawa** [comment]
@gen740 
Ok. In that case, I will quickly check this PR weather it will not break the entire system. If not, I will merge it soon.

**Comment by keisuke-umezawa** [comment]
I confirmed that it also works well in my env. I merge it. Let's add some tests later.

**Pull request merged**